### PR TITLE
Fixed lifecycle twice executing (close #1742)

### LINKED ISSUE #1742: Vuejs: the stories are executed twice
Hi, I detected my components are executed twice, at least when using Vuejs (didn't try with React). I am testing with latest 3.2.8 version. To reproduce just create a simple story like this:

```
<script>
export default {
  created () {
    console.log('should execute once')
  }
}
</script>
````

The component above should be inited once, but is done so twice. Can anyone try and confirm this issue?

### PR DESCRIPTION
**Pull request by kazupon**
## What I did
Fixed #1742 

## How to test
```sh
$ yarn bootstrap --core
$ yarn link
$ cd apps/vue
$ yarn link
$ cd ../../examples/vue-kitchen-sink
$ edit src/stories/Button.vue
```

add the below `created` hook:
```js
created() [
  console.log('call created hook');
}
```

```sh
$ yarn run storybook
```

On the browser,   the below actions
- open inspect panel of devtools
- select `console` tab
- rounted Select [Button] of manager panel on Storybook
- page reload storybook page

After that, in the output of the console, output `call created hook` **once** only

### CODE REVIEW

File: app/vue/src/client/preview/render.js [hunk 1]
```diff
@@ -67,6 +67,8 @@ export function renderMain(data, storyStore) {
     //    https://github.com/storybooks/react-storybook/issues/81
     previousKind = selectedKind;
     previousStory = selectedStory;
+  } else {
```
Comment (ndelangen) [inline app/vue/src/client/preview/render.js]: Is this not also a valid change for react? totally guessing here.

either way this PR is good to merge, as it addresses a very nasty bug for Vue users!

**Review by ndelangen — APPROVED** [review]

File: app/vue/src/client/preview/render.js (hunk 1 above)
Comment (kazupon) [reply app/vue/src/client/preview/render.js]: yup, In the Vue, occure the similar issues.

**Comment by mikemahony** [comment]
Call me crazy but this seems to have regressed? I just ran the above steps to reproduce, this time on `6.2.0-alpha.8` (`next`) and I'm getting `call created hook` twice per storybook page load... 

https://user-images.githubusercontent.com/1740063/102585194-cea6cf80-40bc-11eb-96e6-d4d1a803b302.mov

**Comment by mikemahony** [comment]
Call me crazy but this seems to have regressed? I just ran the above steps to reproduce, this time on `6.2.0-alpha.8` (`next`), and I'm getting `call created hook` twice per storybook page load... 

https://user-images.githubusercontent.com/1740063/102585586-7e7c3d00-40bd-11eb-80cd-f42fdc197899.mov

**Pull request merged**